src/evaluator.ts

```diff
@@ -289,22 +298,92 @@ export class Evaluator {
     return this.findVariation(fc.variations, variation);
   }
 
-  boolVariation(
+  private checkPreRequisite(parent: FeatureConfig, target: Target): boolean {
+    if (parent.prerequisites) {
+      log.info(
+        'Checking pre requisites %s of parent feature %s',
+        parent.prerequisites,
+        parent.feature,
+      );
+
+      for (const pqs of parent.prerequisites) {
+        const preReqFeatureConfig = this.query.getFlag(pqs.feature);
+        if (!preReqFeatureConfig) {
+          log.warn(
+            'Could not retrieve the pre requisite details of feature flag: %s',
+            parent.feature,
+          );
+          return true;
+        }
+
+        // Pre requisite variation value evaluated below
+        const variation = this.evaluateFlag(preReqFeatureConfig, target);
+        log.info(
+          'Pre requisite flag %s has variation %s for target %s',
+          preReqFeatureConfig.feature,
+          variation,
+          target,
+        );
+
+        // Compare if the pre requisite variation is a possible valid value of
+        // the pre requisite FF
+        log.info(
+          'Pre requisite flag %s should have the variations %s',
+          preReqFeatureConfig.feature,
+          pqs.variations,
+        );
+
+        if (!pqs.variations.includes(variation.identifier)) {
+          return false;
+        } else {
+          return this.checkPreRequisite(preReqFeatureConfig, target);
+        }
+      }
+    }
+    return true;
+  }
+
+  private evaluate(
     identifier: string,
     target: Target,
-    defaultValue = false,
+    expected: FeatureConfigKindEnum,
     callback: Callback = undefined,
-  ): boolean {
+  ): Variation | undefined {
     const fc = this.query.getFlag(identifier);
-    if (!fc || fc.kind !== FeatureConfigKindEnum.Boolean) {
-      return defaultValue;
+    if (!fc || fc.kind !== expected) {
+      return undefined;
+    }
+    if (fc.prerequisites) {
+      const prereq = this.checkPreRequisite(fc, target);
+      if (!prereq) {
+        return this.findVariation(fc.variations, fc.offVariation);
+      }
     }
 
     const variation = this.evaluateFlag(fc, target);
     if (variation) {
       if (callback) {
         callback(fc, target, variation);
       }
+      return variation;
+    }
+
+    return undefined;
+  }
+
+  boolVariation(
+    identifier: string,
+    target: Target,
+    defaultValue = false,
+    callback: Callback = undefined,
```

Comment:
If this is `undefined` by default, should `callback` be optional?
```suggestion
    callback?: Callback,
```

Reply:
Done